### PR TITLE
Automated cherry pick of #106906: mount-utils: Detect potential stale file handle

### PR DESCRIPTION
**Pull request by andyzhangx**
Cherry pick of #106906 on release-1.23.

#106906: mount-utils: Detect potential stale file handle

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

#### Does this PR introduce a user-facing change?

```release-note
mount-utils: Detect potential stale file handle
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
/kind bug
/priority important-soon
/sig storage
/triage accepted
/assign @jsafrane 
cc @tsmetana @jingxu97 @Jiawei0227

**Comment by xmudrii** [comment]
Do we also need to cherry-pick this to other release branches or only 1.23 is affected?

**Comment by andyzhangx** [comment]
> Do we also need to cherry-pick this to other release branches or only 1.23 is affected?

@xmudrii sure, I have cherry-picked to 1.21 & 1.22:
https://github.com/kubernetes/kubernetes/pull/107039
https://github.com/kubernetes/kubernetes/pull/107040

**Comment by puerco** [comment]
Hey @andyzhangx thanks for submitting the CPs for the other branches!

We were checking the progress of this cherry-pick but we are forced to wait for the next patch set because we need to merge them all together to keep the branches even.

Let's get the missing approvals and tests ready in #107039 #107040 for the next round.

**Review by justaugustus — APPROVED** [review]
/release-note-none

**Pull request merged**
